### PR TITLE
[Feature] HamiltonianEvolution supported in GPSR with fixed generators

### PR DESCRIPTION
**Pull request by chMoussa**

### CODE REVIEW

File: pyqtorch/matrices.py [hunk 1]
```diff
@@ -67,14 +67,38 @@ def PROJMAT(ket: Tensor, bra: Tensor) -> Tensor:
...
 def parametric_unitary(
-    theta: torch.Tensor, P: torch.Tensor, I: torch.Tensor, batch_size: int  # noqa: E741
+    theta: torch.Tensor,
+    P: torch.Tensor,
+    identity_mat: torch.Tensor,
+    batch_size: int,
+    a: float = 0.5,  # noqa: E741
 ) -> torch.Tensor:
-    cos_t = torch.cos(theta / 2).unsqueeze(0).unsqueeze(1)
+    """Compute the exponentiation of a matrix :math:`P`
```
Comment (jpmoutinho) [inline pyqtorch/matrices.py:76]: exponentiation of a _Pauli_ matrix right?

File: pyqtorch/matrices.py [hunk 2]
```diff
@@ -67,14 +67,38 @@ def PROJMAT(ket: Tensor, bra: Tensor) -> Tensor:
 
 
 def parametric_unitary(
-    theta: torch.Tensor, P: torch.Tensor, I: torch.Tensor, batch_size: int  # noqa: E741
+    theta: torch.Tensor,
+    P: torch.Tensor,
+    identity_mat: torch.Tensor,
```
Comment (jpmoutinho) [inline pyqtorch/matrices.py:72]: Why do you need this as an argument? Can't you use `IMAT` for `pyqtorch.matrices`?

File: pyqtorch/primitive.py
Comment (jpmoutinho) [inline pyqtorch/primitive.py:1]: Seems you have this duplicated from the new `pyqtorch.primitives` module.

**Review by Roland-djee — APPROVED** [review]
LGTM. Not much to add from @jpmoutinho's comments.

File: pyqtorch/matrices.py (hunk 2 above)
Comment (chMoussa) [reply pyqtorch/matrices.py:72]: It is for supporting the device and dtype conversions from the Parametric gate.

**Pull request merged**